static/app/views/settings/components/dataSecrecy/index.tsx

```diff
@@ -0,0 +1,170 @@
+  type DateTimeFieldProps,
+} from 'sentry/components/forms/fields/dateTimeField';
+import Panel from 'sentry/components/panels/panel';
+import PanelAlert from 'sentry/components/panels/panelAlert';
+import PanelBody from 'sentry/components/panels/panelBody';
+import PanelHeader from 'sentry/components/panels/panelHeader';
+import {t, tct} from 'sentry/locale';
+import {useApiQuery} from 'sentry/utils/queryClient';
+import useApi from 'sentry/utils/useApi';
+import useOrganization from 'sentry/utils/useOrganization';
+
+type WaiverData = {
+  accessEnd: string | null;
+  accessStart: string | null;
+};
+
+export default function DataSecrecy() {
+  const api = useApi();
+  const organization = useOrganization();
+
+  const [allowAccess, setAllowAccess] = useState(organization.allowSuperuserAccess);
+  const [allowDate, setAllowDate] = useState<WaiverData>();
+  const [allowDateFormData, setAllowDateFormData] = useState<string>('');
+
+  const {data, refetch} = useApiQuery<WaiverData>(
+    [`/organizations/${organization.slug}/data-secrecy/`],
+    {
+      staleTime: 3000,
+      retry: (failureCount, error) => failureCount < 3 && error.status !== 404,
+    }
+  );
+
+  const hasValidTempAccess =
+    allowDate?.accessEnd && moment().toISOString() < allowDate.accessEnd;
+
+  useEffect(() => {
+    if (data?.accessEnd) {
+      setAllowDate(data);
+      // slice it to yyyy-MM-ddThh:mm format (be aware of the timezone)
+      const localDate = moment(data.accessEnd).local();
+      setAllowDateFormData(localDate.format('YYYY-MM-DDTHH:mm'));
+    }
+  }, [data]);
+
+  const updateAllowedAccess = async (value: boolean) => {
+    try {
+      await api.requestPromise(`/organizations/${organization.slug}/`, {
+        method: 'PUT',
+        data: {allowSuperuserAccess: value},
+      });
+      setAllowAccess(value);
+      addSuccessMessage(t('Successfully updated access.'));
+    } catch (error) {
+      addErrorMessage(t('Unable to save changes.'));
+    }
+  };
+
+  const updateTempAccessDate = async () => {
+    if (!allowDateFormData) {
+      try {
+        await api.requestPromise(`/organizations/${organization.slug}/data-secrecy/`, {
+          method: 'DELETE',
+        });
+        setAllowDate({accessStart: '', accessEnd: ''});
+        addSuccessMessage(t('Successfully removed temporary access window.'));
+      } catch (error) {
+        addErrorMessage(t('Unable to remove temporary access window.'));
+      }
+
+      return;
+    }
+
+    // maintain the standard format of storing the date in UTC
+    // even though the api should be able to handle the local time
+    const nextData: WaiverData = {
+      accessStart: moment().utc().toISOString(),
+      accessEnd: moment.tz(allowDateFormData, moment.tz.guess()).utc().toISOString(),
+    };
+
+    try {
+      await await api.requestPromise(
+        `/organizations/${organization.slug}/data-secrecy/`,
+        {
+          method: 'PUT',
+          data: nextData,
```

Comment:
wait a minute. Is this working? Now that you camelcased the frontend code I don't think passing this directly works.

Reply:
i updated the endpoint to support camel case and now this is working